code/modules/mob/living/carbon/alien/diona/diona_powers.dm

```diff
@@ -185,75 +172,72 @@
 		donor.adjustBruteLoss(2)
-		src.adjustBruteLoss(15)//biting metal hurts!
+		src.adjustBruteLoss(15) //biting metal hurts!
 		return
 
 	else if (isanimal(donor) && (types & TYPE_ORGANIC) && donor.stat != DEAD)
-		src.visible_message("<span class='danger'>[src] bites into [donor.name] and drains some of their blood</span>", "<span class='danger'>You bite into [donor.name] and drain some blood.</span>")
-		to_chat(src, "<span class='danger'>This simple creature has insufficient intelligence for you to learn anything!</span>")
+		src.visible_message(span("danger", "[src] bites into [donor.name] and drains some of their blood."), span("danger", "You bite into [donor.name] and drain some of their blood."))
+		to_chat(src, span("danger", "This simple creature has insufficient intelligence for you to learn anything!"))
 		donor.adjustBruteLoss(4)
 		adjustNutritionLoss(-20)
 		return
 	else if (types & TYPE_WEIRD)
-		src.visible_message("<span class='danger'>[src] attempts to bite into [donor.name] but passes right through it!.</span>", "<span class='danger'>You attempt to sink your fangs into [donor.name] but pass right through it!</span>")
+		src.visible_message(span("danger", "[src] attempts to bite into [donor.name] but passes right through it!."), span("danger", "You attempt to sink your fangs into [donor.name] but pass right through it!"))
 		return
-	else if (donor.is_diona())
-		to_chat(src, span("warning", "You can't sample the DNA of other diona!"))
-		return
-	else if (istype(donor, /mob/living/carbon))
+	else if (iscarbon(donor))
 		//If we get here, it's -probably- valid
 
-		src.visible_message("<span class='danger'>[src] is trying to bite [donor.name]</span>", span("danger", "You start biting [donor.name], you both must stay still!"))
+		src.visible_message(span("danger", "[src] is trying to bite [donor.name]."), span("danger", "You start biting [donor.name], you both must stay still!"))
 		face_atom(get_turf(donor))
-		if (do_mob(src, donor, 40, needhand = 0))
-
+		if(do_mob(src, donor, 40, needhand = FALSE))
 			//Attempt to find the blood vessel, but don't create a fake one if its not there.
 			//If the target doesn't have a vessel its probably due to someone not implementing it properly, like xenos
 			//We'll still allow it
 			var/datum/reagents/vessel = donor.get_vessel(1)
 			var/newDNA
 			var/datum/reagent/blood/B = vessel.get_master_reagent()
 			var/total_blood = B.volume
-			var/remove_amount = (total_blood - 280) * 0.3
-			if (remove_amount > 0)
-				vessel.remove_reagent("blood", remove_amount, 1)//85 units of blood is enough to affect a human and make them woozy
-				adjustNutritionLoss(-remove_amount*0.5)
+			var/remove_amount = total_blood * 0.05 // Take 5% of their blood
```

Comment:
Would it make more sense for this to be 5% of their total blood capacity rather than 5% of remaining blood?

```suggestion
			var/remove_amount = min(species.blood_amount * 0.05, total_blood) // Take 5% of their blood
```